backend/v3/storage/database/dialect/postgres/migration/001_instance_table/up.sql

```diff
@@ -1,6 +1,26 @@
 CREATE TABLE IF NOT EXISTS zitadel.instances(
-    id VARCHAR(100) NOT NULL
-    , PRIMARY KEY (id)
+  id TEXT NOT NULL CHECK (id <> '') PRIMARY KEY,
+  name TEXT NOT NULL CHECK (name <> ''),
+  default_org_id TEXT, -- NOT NULL,
+  iam_project_id TEXT, -- NOT NULL,
+  console_client_id TEXT, -- NOT NULL,
+  console_app_id TEXT, -- NOT NULL,
+  default_language TEXT, -- NOT NULL,
+  created_at TIMESTAMPTZ DEFAULT NOW(),
+  updated_at TIMESTAMPTZ DEFAULT NOW(),
+  deleted_at TIMESTAMPTZ DEFAULT NULL
+);
 
-    , name VARCHAR(100) NOT NULL
-);
\ No newline at end of file
+CREATE OR REPLACE FUNCTION zitadel.set_updated_at()
+RETURNS TRIGGER AS $$
+BEGIN
+  NEW.updated_at := NOW();
+  RETURN NEW;
+END;
+$$ LANGUAGE plpgsql;
+
+CREATE TRIGGER trigger_set_updated_at
+BEFORE UPDATE ON zitadel.instances
+FOR EACH ROW
+WHEN (OLD.* IS DISTINCT FROM NEW.*)
```

Comment:
When we process events we want to have the possibility to set the updated_at of the event so we should allow passed dates.
```suggestion
WHEN (OLD.updated_at IS NOT DISTINCT FROM NEW.updated_at)
```

Reply:
Ya you're right